app/controllers/chat_controller.rb

```diff
@@ -143,7 +143,26 @@ def edit_message
 
   def update_user_last_read
     set_channel_and_chatable
-    set_user_last_read
+    UserChatChannelMembership
+      .where(user: current_user, chat_channel: @chat_channel)
+      .update_all(last_read_message_id: params[:message_id])
+
+    chat_mentions = ChatMention
+      .joins(:notification)
+      .joins(:chat_message)
+      .where(user: current_user)
+      .where(chat_message: { chat_channel_id: @chat_channel.id })
+      .where(notification: { read: false })
+
+    chat_mentions.each do |chat_mention|
+      chat_mention.notification.update(read: true)
```

Comment:
Is this going to do 1 request per mention? 🤔

Reply:
Fixed in #862